newt/toolchain/compiler.go

```diff
@@ -561,7 +561,7 @@ func (c *Compiler) CompileFile(file string, compilerType int) error {
 	return nil
 }
 
-func (c *Compiler) shouldIgnoreFile(file string) bool {
+func (c *Compiler) ShouldIgnoreFile(file string) bool {
 	file = strings.TrimPrefix(file, c.srcDir)
```

Comment:
Won't this leave `file` with a leading `/` character?  It probably doesn't matter in most cases, but if the regex includes a start-of-line-anchor (`^`), then this will be an issue.

I would add:
```
file = strings.TrimLeft(file, "/\\")
```

after the above line.